crates/wdl-engine/src/eval/v1.rs

```diff
@@ -46,6 +47,15 @@ fn write_json_file(path: impl AsRef<Path>, value: &impl Serialize) -> Result<()>
         .with_context(|| format!("failed to write file `{path}`", path = path.display()))
 }
 
+/// A helper struct to associate a value with it's source span.
+#[derive(Debug)]
+pub struct Spanned<T> {
+    /// The value
+    pub value: T,
+    /// The span from the source document.
+    pub span: Span,
+}
```

Comment:
This is generic enough that it should probably go next to `Span`'s definition in `wdl_grammar::Diagnostic`